code/modules/mob/living/basic/trader/trader_ai.dm

```diff
@@ -0,0 +1,120 @@
+/datum/ai_controller/basic_controller/trader
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic,
+	)
+
+	ai_movement = /datum/ai_movement/jps
+	idle_behavior = /datum/idle_behavior/idle_random_walk/not_while_on_target/trader
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/target_retaliate,
+		/datum/ai_planning_subtree/basic_ranged_attack_subtree/trader,
+		/datum/ai_planning_subtree/prepare_travel_to_destination/trader,
+		/datum/ai_planning_subtree/travel_to_point/and_clear_target,
+		/datum/ai_planning_subtree/setup_shop,
+	)
+
+/datum/ai_controller/basic_controller/trader/jumpscare
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/target_retaliate,
+		/datum/ai_planning_subtree/basic_ranged_attack_subtree/trader,
+		/datum/ai_planning_subtree/prepare_travel_to_destination/trader,
+		/datum/ai_planning_subtree/travel_to_point/and_clear_target,
+		/datum/ai_planning_subtree/setup_shop/jumpscare,
+	)
+
+/datum/ai_planning_subtree/basic_ranged_attack_subtree/trader
+	ranged_attack_behavior = /datum/ai_behavior/basic_ranged_attack/trader
+
+/datum/ai_behavior/basic_ranged_attack/trader
+	action_cooldown = 3 SECONDS
+	avoid_friendly_fire = TRUE
+
+///subtree to find our very first customer and set up our shop after walking right into their face
+/datum/ai_planning_subtree/setup_shop
+	/// What do we do in order to offer our deals?
+	var/datum/ai_behavior/setup_shop/setup_shop_behavior = /datum/ai_behavior/setup_shop
+
+/datum/ai_planning_subtree/setup_shop/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	//if we already have a shop spot, return
+	if(controller.blackboard_key_exists(BB_SHOP_SPOT))
+		return
+
+	//if we don't have a costurmer to greet, look for one
+	if(!controller.blackboard_key_exists(BB_FIRST_CUSTOMER))
+		controller.queue_behavior(/datum/ai_behavior/find_and_set/conscious_person, BB_FIRST_CUSTOMER, /mob/living/carbon/human)
+		return
+
+	//we have our first customer, time to tell them about incredible deals
+	controller.queue_behavior(setup_shop_behavior, BB_FIRST_CUSTOMER)
+	return SUBTREE_RETURN_FINISH_PLANNING
+
+///The ai will create a shop the moment they see a potential costumer
+/datum/ai_behavior/setup_shop
+
+/datum/ai_behavior/setup_shop/setup(datum/ai_controller/controller, target_key)
+	. = ..()
+	var/obj/target = controller.blackboard[target_key]
+	if(QDELETED(target))
+		return FALSE
```

Comment:
```suggestion
	var/obj/target = controller.blackboard[target_key]
	return !QDELETED(target)
```
I know we have the impulse to _always_ call super but the super call here just returns TRUE, doing it this way is more explicit about what we want.